Document Atomics.waitAsync

Elchi3 opened the pull request:
I'm not really an expert on shared memory, but this is at least a minimal page on Atomics.waitAsync.

Spec: https://tc39.es/proposal-atomics-wait-async/
Explainer: https://github.com/tc39/proposal-atomics-wait-async/blob/master/PROPOSAL.md

Fixes https://github.com/mdn/content/issues/2005

Elchi3 committed:
Document Atomics.waitAsync

teoli2003 commented on files/en-us/web/javascript/reference/global_objects/atomics/waitasync/index.md:
````diff
@@ -0,0 +1,88 @@
+// { async: false, value: "ok" }
+```
+
+If it isn't resolving to "ok", the value in the shared memory location wasn't the expected on (the promise will resolve to "not-equal") or the timeout was reached (the promise will resolve to "time-out"). 
````
Nit:
```suggestion
If it isn't resolving to `"ok"`, the value in the shared memory location wasn't the expected  (the promise will resolve to `"not-equal"`) or the timeout was reached (the promise will resolve to `"time-out"`). 
```

teoli2003 commented:
Is there a bcd PR?

Elchi3 committed:
Update files/en-us/web/javascript/reference/global_objects/atomics/waitasync/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Elchi3 commented:
> Is there a bcd PR?

Yes, I updated all of the SharedArrayBuffer data in https://github.com/mdn/browser-compat-data/pull/16047 and added this there.

teoli2003 reviewed (approved):
👍
